src/components/SharedComponents/ObservationViews/hooks/useUploadObservations.js

```diff
@@ -2,12 +2,18 @@
 
```

Comment:
Is uploading working consistently with `RealmProvider`? I tried uploading three local observations (two from the camera and one from photo library), and only one made it to staging, without a photo attached.

Reply:
Definitely found and fixed some bugs in recent commits.